gensim/models/word2vec.py

```diff
@@ -509,9 +510,16 @@ def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None, progress_
         Each sentence must be a list of unicode strings.
 
         """
+        for sentence in sentences:
+            if hasattr(sentence, "words"):
+                if isinstance(sentence.words, string_types):
+                    warnings.warn("Sentences should be a list of unicode strings. True string provided.")
+                    break
         self.scan_vocab(sentences, progress_per=progress_per, trim_rule=trim_rule)  # initial survey
         self.scale_vocab(keep_raw_vocab=keep_raw_vocab, trim_rule=trim_rule)  # trim by min_count & precalculate downsampling
         self.finalize_vocab()  # build tables & arrays
+        self.check_vocab() # raise warning if vocab consists of all single characters
```

Comment:
This check seems redundant given the earlier check: the main way someone would wind up with a single-char vocab is if they'd passed simple strings. (If they've already been warned about that, no need to do this check.) 

But also, some will want to do the scan_vocab/scale_vocab/finalize_vocab steps themselves (in lieu of build_vocab), to have more control – and a user doing that should still get whatever checks we deem helpful. So a check like this (if it survives) would better happen within one of those methods, at the earliest possible time the anomaly can be detected.